charts/kong/templates/_helpers.tpl

```diff
@@ -53,76 +53,59 @@ Create the name of the service account to use
+    {{- $httpListen = append $httpListen (printf "0.0.0.0:%d" (int64 .http.containerPort)) -}}
+    {{- range $param := .http.parameters }}
+      {{- $httpListen = append $httpListen $param -}}
+    {{- end -}}
 
-{{- end -}}
+    {{- $unifiedListen = append $unifiedListen ($httpListen | join " ") -}}
+  {{- end -}}
 
-{{/*
-Create the KONG_ADMIN_GUI_LISTEN value string
-*/}}
-{{- define "kong.kongManagerListenValue" -}}
+  {{- if .tls.enabled -}}
+    {{- $tlsListen = append $tlsListen (printf "0.0.0.0:%d ssl" (int64 .tls.containerPort)) -}}
+    {{- range $param := .tls.parameters }}
+      {{- $tlsListen = append $tlsListen $param -}}
+    {{- end -}}
 
-{{- if and .Values.manager.http.enabled .Values.manager.tls.enabled -}}
-   0.0.0.0:{{ .Values.manager.http.containerPort }},0.0.0.0:{{ .Values.manager.tls.containerPort }} ssl
-{{- else -}}
-{{- if .Values.manager.http.enabled -}}
-   0.0.0.0:{{ .Values.manager.http.containerPort }}
-{{- end -}}
-{{- if .Values.manager.tls.enabled -}}
-   0.0.0.0:{{ .Values.manager.tls.containerPort }} ssl
-{{- end -}}
-{{- end -}}
-
-{{- end }}
+    {{- $unifiedListen = append $unifiedListen ($tlsListen | join " ") -}}
+  {{- end -}}
 
-{{/*
-Create the KONG_PORTAL_GUI_LISTEN value string
-*/}}
-{{- define "kong.kongPortalListenValue" -}}
-
-{{- if and .Values.portal.http.enabled .Values.portal.tls.enabled -}}
-   0.0.0.0:{{ .Values.portal.http.containerPort }},0.0.0.0:{{ .Values.portal.tls.containerPort }} ssl
-{{- else -}}
-{{- if .Values.portal.http.enabled -}}
-   0.0.0.0:{{ .Values.portal.http.containerPort }}
-{{- end -}}
-{{- if .Values.portal.tls.enabled -}}
-   0.0.0.0:{{ .Values.portal.tls.containerPort }} ssl
-{{- end -}}
+  {{- $listenString := ($unifiedListen | join ", ") -}}
+  {{- if eq (len $listenString) 0 -}}
+    {{- $listenString = "off" -}}
+  {{- end -}}
+  {{- $listenString -}}
 {{- end -}}
 
-{{- end }}
 
 {{/*
-Create the KONG_PORTAL_API_LISTEN value string
+Return the local admin API URL, preferring HTTPS if available
 */}}
-{{- define "kong.kongPortalApiListenValue" -}}
-
-{{- if and .Values.portalapi.http.enabled .Values.portalapi.tls.enabled -}}
-   0.0.0.0:{{ .Values.portalapi.http.containerPort }},0.0.0.0:{{ .Values.portalapi.tls.containerPort }} ssl
-{{- else -}}
-{{- if .Values.portalapi.http.enabled -}}
-   0.0.0.0:{{ .Values.portalapi.http.containerPort }}
+{{- define "kong.adminLocalURL" -}}
+  {{- if .Values.admin.containerPort -}} {{/* TODO: Remove legacy admin behavior */}}
+    {{- if .Values.admin.useTLS -}}
+https://localhost:{{ .Values.admin.containerPort }}
+    {{- else -}}
+http://localhost:{{ .Values.admin.containerPort }}
+    {{- end -}}
+  {{- else -}}
+    {{- if .Values.admin.tls.enabled -}}
+https://localhost:{{ .Values.admin.tls.containerPort }}
+    {{- else if .Values.admin.http.enabled -}}
+http://localhost:{{ .Values.admin.http.containerPort }}
+    {{- else -}}
+http://localhost:9999 # You have an ingress controller enabled, but no admin listens!
```

Comment:
Can we rephrase this as: "Ingress Controller will not work because there are no admin listeners enabled in Kong."?